components/composer/src/lib/utils.ts

```diff
@@ -16,6 +16,24 @@ export const debounce = (
     timeout = setTimeout(later, wait);
   };
 };
+
 export const isValidEmail = (email: string): boolean => {
   return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
 };
+
+export const calculateMessageTempSnippet = (body: string): string => {
+  if (!body) return "";
+  return (
+    body
+      // Remove style and script tag and content
+      .replace(/<style|script([\S\s]*?)>([\S\s]*?)<\/style|script>/g, "")
+      // Remove all tags and keep content / Remove all HTML entities
+      ?.replace(/<\/?[^>]+(>|$)|(&.+?;)/g, "")
+      // Remove extra spaces
+      ?.replace(/\s+/g, " ")
+      // Remove leading and trailing space
+      ?.trim()
+      // Add ... for longer message
+      ?.substring(0, 190) + (body.length > 190 ? "..." : "")
```

Comment:
@yifanplanet cold we instead use the same library we're using in the HTML body - `dompurify`? 
https://github.com/cure53/DOMPurify

Safer + easier to work with

Reply:
@AaronDDM Updated to code to use DOMPurify for a safer strategy agains xss.

Reply:
Amazing thanks @yifanplanet